### PR TITLE
Improving title discovery

### PR DESCRIPTION
**Pull request by gsambrotta**
Create a visual feedback for user on notifications when they grant a badge that they could use as title.
Allow user to change title directly from username/badge page.

Code Modification:
- pass badge_title attribute to notification-item in order to check if notification badge give feedbacks to user
- Make badge-title as component so it could be reuse in other part of the app, like badge/show

Issue: https://meta.discourse.org/t/improving-discovery-of-the-badge-title-feature/27372

### CODE REVIEW

**Comment by ZogStriP** [comment]
Thanks! Would you mind adding some before/after screenshots?

**Comment by Toxu-ru** [comment]
Search before and after
toxu

**Comment by erlend-sh** [comment]
> Create a visual feedback for user on notifications when they grant a badge that they could use as title.

This part of the PR has [been rejected](https://meta.discourse.org/t/improving-discovery-of-the-badge-title-feature/27372/17). But I think the second part is welcome:

> Allow user to change title directly from username/badge page.

@coding-horror can we merge this if the rejected notification change is stripped out, while the badge page tweak is left in?

**Comment by coding-horror** [comment]
Sure that is fine, the badge page is a better place for this UI anyways

**Comment by gsambrotta** [comment]
@erlend-sh @coding-horror 
Just pushed the changes that remove the visual feedback.
Here the screenshots of the modified part
schermata 2017-06-14 alle 22 50 30
schermata 2017-06-14 alle 22 50 41

**Comment by coding-horror** [comment]
👍 🎉

**Comment by SamSaffron** [comment]
@eviltrout can you review and merge if it looks good?

File: app/assets/javascripts/discourse/components/badge-title.js.es6 [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+  actions: {
+    save() {
+      this.setProperties({ saved: false, saving: true });
+
+      ajax(this.get('user.path') + "/preferences/badge_title", {
+        type: "PUT",
+        data: { user_badge_id: this.get('selectedUserBadgeId') }
+      }).then(() => {
+        this.setProperties({
+          saved: true,
```
Comment (eviltrout) [inline app/assets/javascripts/discourse/components/badge-title.js.es6]: You should declare the attributes you will be using in the component with default values. Looks like `saved: false, saving:false` - Javascript is much faster when it knows the shape of objects in advance

File: app/assets/javascripts/discourse/routes/badges-show.js.es6 [hunk 2]
```diff
@@ -28,11 +28,22 @@ export default Discourse.Route.extend({
...
-      this.userBadges = userBadges;
+  	const userBadgesGrant = UserBadge.findByBadgeId(model.get("id"), {username}).then(userBadges => {
+      this.userBadgesGrant = userBadges;
     });
+
+  	const userBadgesAll = UserBadge.findByUsername(username).then(userBadges => {
+      this.userBadgesAll = userBadges;
+    });
+
+    const promises = {
```
Comment (eviltrout) [inline app/assets/javascripts/discourse/routes/badges-show.js.es6]: The formatting in this file is off. Can you please correct it?

**Comment by SamSaffron** [comment]
@nbianca  can you help me merge this in? we are going to need a new PR that addresses the problems and conflict.

**Comment by SamSaffron** [comment]
@gsambrotta thanks so much, this is now merged via https://github.com/discourse/discourse/pull/5049

**Pull request closed without merging**